checker/src/org/checkerframework/checker/index/inequality/LessThanAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,263 @@
+            }
+
+            return subList.containsAll(superList);
+        }
+
+        @Override
+        public AnnotationMirror leastUpperBound(AnnotationMirror a1, AnnotationMirror a2) {
+            if (isSubtype(a1, a2)) {
+                return a2;
+            } else if (isSubtype(a2, a1)) {
+                return a1;
+            }
+
+            List<String> a1List = getLessThanExpressions(a1);
+            List<String> a2List = getLessThanExpressions(a2);
+            List<String> lub = new ArrayList<>(a1List);
+            lub.retainAll(a2List);
+
+            return createLessThanQualifier(lub);
+        }
+
+        @Override
+        public AnnotationMirror greatestLowerBound(AnnotationMirror a1, AnnotationMirror a2) {
+            if (isSubtype(a1, a2)) {
+                return a1;
+            } else if (isSubtype(a2, a1)) {
+                return a2;
+            }
+
+            List<String> a1List = getLessThanExpressions(a1);
+            List<String> a2List = getLessThanExpressions(a2);
+            List<String> glb = new ArrayList<>(a1List);
+            glb.addAll(a2List);
+
+            return createLessThanQualifier(glb);
+        }
+    }
+
+    /** @return Is left less than right? */
+    public boolean isLessThan(Tree left, String right) {
+        AnnotatedTypeMirror leftATM = getAnnotatedType(left);
+        if (leftATM.getAnnotations().size() != 1) {
+            return false;
+        }
+        List<String> expressions =
+                getLessThanExpressions(leftATM.getAnnotations().iterator().next());
+        if (expressions == null) {
+            return true;
+        }
+        return expressions.contains(right);
+    }
+
+    /** @return {@code smaller < bigger}, using information from the Value Checker */
+    public boolean isLessThanByValue(Tree smaller, String bigger, TreePath path) {
+        Long smallerValue = IndexUtil.getMinValue(smaller, getValueAnnotatedTypeFactory());
+        if (smallerValue == null) {
+            return false;
+        }
+
+        OffsetEquation offsetEquation = OffsetEquation.createOffsetFromJavaExpression(bigger);
+        if (offsetEquation.isInt()) {
+            // bigger is an int literal
+            return smallerValue < offsetEquation.getInt();
+        }
+        // If bigger is "expression + literal", then smaller < expression + literal
+        // can be reduced to smaller - literal < expression + literal - literal
+        smallerValue = smallerValue - offsetEquation.getInt();
+        offsetEquation =
+                offsetEquation.copyAdd(
+                        '-', OffsetEquation.createOffsetForInt(offsetEquation.getInt()));
+
+        long minValueOfBigger = getMinValueFromString(offsetEquation.toString(), smaller, path);
+        return smallerValue < minValueOfBigger;
+    }
+
+    /** Returns the minimum value of {@code expressions} at {@code tree}. */
+    private long getMinValueFromString(String expression, Tree tree, TreePath path) {
```

Comment:
Maybe this method should just throw a `FlowExpressionParseException` instead of silently hiding them?